src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -414,15 +414,23 @@ private void PopulateSourceRepositoryCache(ListPackageArgs listPackageArgs)
             {
                 foreach (var topLevelPackage in frameworkPackages.TopLevelPackages)
                 {
-                    var matchingPackage = packageMetadata.Where(p => p.Key.Equals(topLevelPackage.Name, StringComparison.OrdinalIgnoreCase)).First();
+                    var matchingPackage = packageMetadata.Where(p => p.Key.Equals(topLevelPackage.Name, StringComparison.OrdinalIgnoreCase)).FirstOrDefault();
```

Comment:
Off topic for this PR, but I wonder if [this dictionary can simply be created with this comparer](https://github.com/NuGet/NuGet.Client/pull/5427/files#diff-60e044f4a652f6a16cc3c89aedc2cbcf13361c672c237a95268a3e95905340e5R278), like so:

```C#
var packageMetadataById = new Dictionary<string, List<IPackageSearchMetadata>>(capacity: allPackages.Count);
```

Then, you could simply do a dictionary lookup:

```
if (packageMetadata.TryGetValue(topLevelPackage.Name, out List<string>? versions))
```

Reply:
I discovered that this dictionary is initialized that way in https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs#L278. I pushed a new commit; do you think it addresses your suggestion?